### DISCUSSION TITLE
Support adding API Deepl and Chatgpt

### DISCUSSION THREAD

**Discussion by nolilis** [xulihang/ImageTrans-docs#551] [General]
I am a new user. I tried adding API keys but they don't work Deepl and chatgpt
![image](https://github.com/xulihang/ImageTrans-docs/assets/76251604/d78a9cb9-41d6-4e9e-9339-f7135c9bdfff)
![image](https://github.com/xulihang/ImageTrans-docs/assets/76251604/1afbeef7-0d21-475c-9b76-fccb8e82ba7f)

**Comment by xulihang** [comment]
How do you know that they do not work? I don't see you calling them.

You can enable them in the machine translation preferences and see them in the Transation Assist Tab or use "Translate all text areas" to use them.

If you cannot get the translation, check whether you've set up a correct language pair in the project settings.

In addition, you can use `ImageTrans.bat` to start the program and check out the detailed logs which contains the error messages.

**Reply by nolilis** [reply]
everything is fine but I can't use deepl and GPT, is it because I installed it wrong? If  use Baidu, can use it
![image](https://github.com/xulihang/ImageTrans-docs/assets/76251604/3ed182ce-53f7-4716-ac3c-3c591b7b1ab1)
![image](https://github.com/xulihang/ImageTrans-docs/assets/76251604/241950f2-8852-4f77-bf3c-9bdc32273a78)

**Reply by xulihang** [reply]
Use ImageTrans.bat to start the program and check out the detailed logs which contain the error messages.

**Reply by nolilis** [reply]
I use deepl . This message appears, what should I do to fix it?
![image](https://github.com/xulihang/ImageTrans-docs/assets/76251604/3ae61d17-1697-4122-93d8-7041b38072d7)

**Reply by xulihang** [reply]
Open project settings to set the language pair. If you need to translate Chinese, the language code should be zh.

**Reply by nolilis** [reply]
Thank you, I did it. there's still one more problem . when I select balloon detecte . they don't work
![image](https://github.com/xulihang/ImageTrans-docs/assets/76251604/cc48cbb5-99b1-4c44-8889-0d5cb66f0445)
![image](https://github.com/xulihang/ImageTrans-docs/assets/76251604/ebece6cd-cd71-4784-9052-37a39de7a55d)

**Reply by xulihang** [reply]
The default balloon detection has a daily usage limit. You can use the heuristic detection instead or other OCRs.
